src/cascadia/TerminalApp/AppLogic.cpp

```diff
@@ -852,7 +297,8 @@ namespace winrt::TerminalApp::implementation
             }
 
             _settings = std::move(newSettings);
-            hr = _warnings.empty() ? S_OK : S_FALSE;
+
+            hr = (_warnings.size()) == 0u ? S_OK : S_FALSE;

```

Comment:
nit: why the move from `empty` to `size==0`? Leonard has remarked on this on past PRs, but I can't recall exactly

Reply:
It was lost in the `vector`->`IVector`->`vector` iteration